src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -1143,16 +1143,21 @@ internal void RecordAndDeleteModels(List<ModelBase> models)
                         var node = model as NodeModel;
                         Debug.Assert(Nodes.Contains(node));
 
+                        bool silentFlag = node.RaisesModificationEvents;
+                        node.RaisesModificationEvents = false;
```

Comment:
I was thinking the same, but that's "in future", which means the solution can wait (also introducing a fix here will deviate from the master branch). And the right solution, as @ke-yu rightly point out in this previous PR, is to introduce the concept of a *transaction* to Dynamo, at the end of each transaction notifications are sent/handled.